bicycleparameters/cycle_app.py

```diff
@@ -213,7 +219,14 @@
 
 # creates generic set of Benchmark parameters
 
-
+@app.callback(
+    dash.dependencies.Output('download-button', 'href'),
+    [dash.dependencies.Input('bike-dropdown', 'value')])
+def update_download_link(filter_value):
+    dff = bicycle.eig(Benchmark)
```

Comment:
It doesn't look like you are calculating the eigenvalues correctly:

https://pythonhosted.org/BicycleParameters/bicycleparameters.html#bicycleparameters.main.Bicycle.eig